jax/_src/debugger/colab_debugger.py

```diff
@@ -136,18 +141,16 @@ def update_frame(self, frame):
     self.frame = frame
     lineno = self.frame.lineno or None
     filename = self.frame.filename.strip()
-    if inspect.getmodulename(filename):
-      if filename not in self._file_cache:
-        try:
-          with open(filename) as fp:
-            self._file_cache[filename] = fp.read()
-          source = self._file_cache[filename]
-          highlight = lineno
-          linenostart = 1
-        except FileNotFoundError:
-          source = "\n".join(frame.source)
-          highlight = min(frame.offset + 1, len(frame.source) - 1)
-          linenostart = lineno - frame.offset
+    if inspect.getmodulename(filename) and filename not in self._file_cache:
+      try:
+        with open(filename) as fp:
+          self._file_cache[filename] = fp.read()
+      except FileNotFoundError:
+        pass
+    if filename in self._file_cache:
+      source = self._file_cache[filename]
```

Comment:
This is a change in program logic – this block now executes even if `inspect.getmodulename(filename)` is False. I'm not sure of the ramifications of that here: is that an intended change?

Reply:
I think this was a bug in the original implementation. 

If the `filename` is already cached (and it could've only been cached if `inspect.getmodulename(filename)` returned True AFAICT), this method would raise `NotImplementedError`.